### ISSUE TITLE
Initialization vector (nstart) for all pagerank variants

### ISSUE THREAD

**Issue by lebiathan** [networkx/networkx#3182]
Hi,

I've been using networkx for a brief time now, but noticed that when it comes to the implemented [PageRank variants](https://networkx.github.io/documentation/stable/reference/algorithms/link_analysis.html) only the Python version accepts an initialization vector (**nstart**) as a parameter.

From what I've read - but haven't tried it out so far - the scipy and numpy implementations should be faster than the pure python version, but not being able to provide the initialization vector takes them off the table for a variety of cases.

I was wondering if that omission of **nstart** is by design and there's some underlying reason that I don't fully understand.

As I didn't find a related discussion / request in the past issues, I opened a new one. I'd be happy to submit a PR that adds the **nstart** parameters to the other methods, treating it in the same was as with the Python one.

George

**Comment by dschult** [comment]
I believe it hasn't been implemented because of ease rather than design. 
It would be great to have that addition to NetworkX.  Thanks!

**Comment by hagberg** [comment]
The 'nstart' parameter is, as you say, the initialization vector for the power iteration algorithm to compute the pagerank vector.    The vector is sometimes chosen to be constant (e.g. normalized vector of all 1's), or random.   The choice will have an effect on the time to convergence since e.g. if you start very close to the answer you will need very few iterations.    

I don't see any reason why we couldn't add the ability to specify the starting vector for the pagegrank_scipy algorithm.  

The pagerank_numpy algorithm uses a direct eigenvalue solver (not an iterative method) so it doesn't apply there.

**Comment by lebiathan** [comment]
Thanks for the prompt replies!

Given @hagberg 's comment, it makes sense that the 'nstart' parameter wouldn't be of any help to ```pagerank_numpy()``` - to be honest, I actually skipped that method while looking into the code. Do you think it would make sense to, somehow, inform the user of this? I imagine that others might be as surprised as I was, seeing that only one of the methods supports the 'nstart' parameter.


A question on the "logistics" side of things. While following your 'how to' file about contributions, I ran into the following error when executing one of the commands:
```
$ git pull upstream master
Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```
I'm not sure if the instructions are slightly outdated or I've missed a step along the way.

In any case, I've opened a pull request with the modified code (). Let me know what more I should provide / improve!

Thanks,
George

**Comment by dschult** [comment]
Fixed in #3183
